honeycomb/honeycomb.go

```diff
@@ -101,11 +111,34 @@ func (e *Exporter) ExportSpan(data *trace.SpanData) {
 	hs := honeycombSpan(data)
 	ev.Add(hs)
 
-	// Add an event field for each attribute
-	// Hrm. Seems like attributes are trying to tell us something about the type of span
+	// TODO: (akvanhar) do something about the attributes
 	// ev.Add(data.Attributes)
+
+	// We send these message events as 0 duration spans
 	for _, a := range data.MessageEvents {
-		ev.Add(a)
+		messageEv := e.Builder.NewEvent()
+		if e.ServiceName != "" {
+			messageEv.AddField("service_name", e.ServiceName)
+		}
+
+		// TODO: (akvanhar) We can't export these as attributes are not yet exported fields
+		// if len(a.attributes) != 0 {
+		// 	for key, value := range a.attributes {
+		// 		messageEv.AddField(key, value)
+		// 	}
+		// }
+
+		// TODO: (akvanhar) This doesn't actually do anything as libhoney's Add will only add exported fields
+		messageEv.Add(a)
```

Comment:
This is a bit of a placeholder for now as a use for `a`. It will be removed when we can access the `Event` type.